xbmc/filesystem/ZipFile.cpp

```diff
@@ -305,7 +305,7 @@ ssize_t CZipFile::Read(void* lpBuf, size_t uiBufSize)
   {
     uLong iDecompressed = 0;
     uLong prevOut = m_ZStream.total_out;
-    while (((int)iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
+    while ((iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
```

Comment:
Wouldn't it be safe as it's sign safe and the comparison should be safe as well between uint32_t and uint64_t (implicit)?